### PR TITLE
handle missing translations for min and max validations

### LINKED ISSUE #7305: No translation for min and max validation errors
From https://github.com/mraible/21-points/issues/33

### PR DESCRIPTION
**Pull request by ruddell**
Fix #7305

 - Maps Min/Max/DecimalMin/DecimalMax to the Size i18n key on the client.  
 - Adds test for `alert-error.component.ts` 

Please make sure the below checklist is followed for Pull Requests.

- [x] [Travis tests](https://travis-ci.org/jhipster/generator-jhipster/pull_requests) are green
- [x] Tests are added where necessary
- [x] Documentation is added/updated where necessary
- [x] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/master/CONTRIBUTING.md) are followed

### CODE REVIEW

**Comment by vishal423** [comment]
@ruddell , can this conversion be done on client side as I believe the issue is due to client limitation to handle this out. API contract shouldn't be driven by client implementation, rather it's other way around. As you also said, elegant solution would be to add missing translations on client side to sync with API error contract.

**Comment by ruddell** [comment]
@vishal423 How does the updated PR look?  I added the Min/Max i18n keys for all languages and made DecimalMin and DecimalMax use those keys instead of Size.  We could add translations for those two as well but it would be the same translation

**Comment by vishal423** [comment]
@ruddell, sorry, but, looking at changes I don't see value add of adding new keys as conveyed message is similar to that of 'size' key. Instead, can this mapping be done on client itself to map to 'size' key instead of min/max/? That way server contract is respected and client side validation messages are reused.

**Comment by deepu105** [comment]
@mraible @vishal423 would you be able to test this

**Comment by ruddell** [comment]
To test, you have to remove the `jhiMin`/`jhiMax` Angular directives or `min`/`max` React validation on the entity update dialog (otherwise you just get client validation).  You can use the below JDL:

```
entity Preferences {
    goal Integer min(10) max(20)
    exactGoal Double min(10) max(20)
}
```

File: generators/client/templates/angular/src/test/javascript/spec/app/shared/alert/alert-error.component.spec.ts.ejs [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+        let eventManager: JhiEventManager;
+
+        beforeEach(async(() => {
+            TestBed.configureTestingModule({
+                imports: [<%=angularXAppName%>TestModule<% if (enableTranslation) { %>, TranslateModule.forRoot()<% } %>],
+                declarations: [<%=jhiPrefixCapitalized%>AlertErrorComponent],
+                providers: [
+                    JhiEventManager,
+                    {
+                        provide: JhiAlertService,
```
Comment (vishal423) [inline generators/client/templates/angular/src/test/javascript/spec/app/shared/alert/alert-error.component.spec.ts.ejs]: do we require prefix variable here?

**Comment by vishal423** [comment]
Overall change looks good to me, however, wouldn't be able to test this out.

File: generators/client/templates/angular/src/test/javascript/spec/app/shared/alert/alert-error.component.spec.ts.ejs (hunk 1 above)
Comment (ruddell) [reply generators/client/templates/angular/src/test/javascript/spec/app/shared/alert/alert-error.component.spec.ts.ejs]: `JhiEventManager` and `JhiAlertService` come from `ng-jhipster` so it always uses the `jhi` prefix

File: generators/client/templates/react/src/main/webapp/app/config/notification-middleware.ts.ejs [hunk 2]
```diff
@@ -101,6 +101,9 @@ export default () => next => action => {
                 const fieldErrors = data.fieldErrors;
                 for (i = 0; i < fieldErrors.length; i++) {
                   const fieldError = fieldErrors[i];
+                  if (['Min', 'Max', 'DecimalMin', 'DecimalMax'].includes(fieldError.message)) {
```
Comment (vishal423) [inline generators/client/templates/react/src/main/webapp/app/config/notification-middleware.ts.ejs]: would it be possible to add unit test for react as well?

Comment (ruddell) [reply generators/client/templates/react/src/main/webapp/app/config/notification-middleware.ts.ejs]: Done

**Comment by ruddell** [comment]
Claimed the bug bounty here: https://opencollective.com/generator-jhipster/expenses/4421

**Pull request merged**